api/src/middleware/authorization.js

```diff
@@ -0,0 +1,55 @@
+const { ERR_AUTH_FAILED, ERR_NOT_AUTHORIZED } = require('../utils/constants');
+const { sendResponse } = require('../utils/response');
+const { models } = require('../models/index.js');
+
+const requireLanguageAuthorization = async (req, res, next) => {
+  var current_language = '';
+  if (req.body.course_id) {
+    current_language = req.body.course_id;
+  } else if (req.query.course_id) {
+    current_language = req.query.course_id;
+  } else if (req.params.id) {
+    current_language = req.params.id;
+  } else {
+    return sendResponse(res, 401, ERR_AUTH_FAILED);
+  }
+  var ObjectId = require('mongoose').Types.ObjectId;
+  if (!ObjectId.isValid(current_language)) {
+    next();
+    return;
+  }
+  const courseExists = await models.Course.findById(current_language);
```

Comment:
I would wrap this line in a try catch since the models.Course.findById() might throw an error and return a 500 code. Probably can just include this in the try catch below.